### ISSUE TITLE
Supply - add automatic AAB path detection after gradle action

### ISSUE THREAD

**Issue by kusan-thana** [fastlane/fastlane#13171]
### Feature Request
Supply should look for Actions::SharedValues:GRADLE_AAB_OUTPUT_PATH automatically like it did with apk (APK_OUTPUT_PATH).

#### Motivation Behind Feature
It's a bit of a regression since deploying with apk was easier to configure.
```

+-------------------------+-----------------------------------------------------------+
|                             Summary for supply 2.102.0                              |
+-------------------------+-----------------------------------------------------------+
| track                   | internal                                                  |
| package_name            | toto                                         |
| json_key                |toto.json |
| skip_upload_apk         | false                                                     |
| skip_upload_aab         | false                                                     |
| skip_upload_metadata    | false                                                     |
| skip_upload_images      | false                                                     |
| skip_upload_screenshots | false                                                     |
| validate_only           | false                                                     |
| check_superseded_tracks | false                                                     |
| timeout                 | 300                                                       |
+-------------------------+-----------------------------------------------------------+

+-----------------------------+-------------------------------------------------------------------------------------------------+
|                                                         Lane Context                                                          |
+-----------------------------+-------------------------------------------------------------------------------------------------+
| DEFAULT_PLATFORM            | android                                                                                         |
| PLATFORM_NAME               | android                                                                                         |
| LANE_NAME                   | android deploy_internal                                                                         |
| ANDROID_NEW_VERSION_CODE    | 365                                                                                             |
| GRADLE_BUILD_TYPE           | alpha                                                                                           |
| GRADLE_ALL_APK_OUTPUT_PATHS | []                                                                                              |
| GRADLE_ALL_AAB_OUTPUT_PATHS | ["/builds/R1130/VKO/toto/Application_Android/app/build/outputs/bundle/alpha/app.aab"] |
| GRADLE_AAB_OUTPUT_PATH      | /builds/R1130/VKO/toto/Application_Android/app/build/outputs/bundle/alpha/app.aab     |
+-----------------------------+-------------------------------------------------------------------------------------------------+
[15:28:37]: No local metadata, apks, aab, or track to promote were found, make sure to run `fastlane supply init` to setup supply

+------+--------------------------------------------+-------------+
|                        fastlane summary                         |
+------+--------------------------------------------+-------------+
| Step | Action                                     | Time (in s) |
+------+--------------------------------------------+-------------+
| 1    | update_fastlane                            | 5           |
| 2    | default_platform                           | 0           |
| 3    | Switch to android latest_build_number lane | 0           |
| 4    | google_play_track_version_codes            | 4           |
| 5    | google_play_track_version_codes            | 4           |
| 6    | google_play_track_version_codes            | 2           |
| 7    | android_set_version_code                   | 0           |
| 8    | gradle                                     | 13          |
| 9    | gradle                                     | 82          |
| 💥   | upload_to_play_store                       | 2           |
+------+--------------------------------------------+-------------+

[15:28:37]: fastlane finished with errors

[!] No local metadata, apks, aab, or track to promote were found, make sure to run `fastlane supply init` to setup supply
ERROR: Job failed: exit code 1
```

**Comment by janpio** [comment]
Regressions are to be used for when something worked, but not does not work again.

What exactly is being run in the output you pasted above?

Your feature request is valid though, `supply`/`upload_to_play_store` only takes the APK vars:
https://github.com/fastlane/fastlane/blob/2e28e6ff975df87096bbb802dfcc25d510473389/fastlane/lib/fastlane/actions/upload_to_play_store.rb#L8-L17
(although the way this is build is a bit strange anyway, as it only works in the action and not command line. Not sure if this is different in other actions/tools)

**Comment by janpio** [comment]
> Yep, sorry for the regression, didn't expect the bot!

All good, I removed the label and we are now handling this as a feature request.

> What's the difference between action and command line?

`upload_to_play_store` also has a second name: `supply`. You can run it in a lane, like you are doing, or just execute `fastlane supply` on the command line as it is one of a limited sets of "special" commands. What I noticed is that the code, that gets the APK path from `gradle` only works in the action. But now explaining it to you it makes semi sense, as you can't run `gradle` on the command line as it is not one of the special ones.

**Comment by jschmid** [comment]
@kusan-thana if you haven't see it, I fixed this in #13496
